supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiAsyncBackendJobExecutionActor.scala

```diff
@@ -482,6 +483,41 @@ class PipelinesApiAsyncBackendJobExecutionActor(override val standardParams: Sta
 
         val enableSshAccess = workflowOptions.getBoolean(WorkflowOptionKeys.EnableSSHAccess).toOption.contains(true)
 
+        val dockerImageWithDigest = jobDockerImage
+        val dockerImageAsSpecifiedByUser = runtimeAttributes.dockerImage
+        val dockerImageCacheDiskOpt =
+          jesAttributes
+            .dockerImageToCacheDiskImageMappingOpt
+            .flatMap(_.get(dockerImageAsSpecifiedByUser))
+            .filter { cachedDockerImageDigestAndDiskName =>
+              val hashStartingPositionInActualDockerImage = dockerImageWithDigest.indexOf('@')
+              if (hashStartingPositionInActualDockerImage != -1) {
+                val actualDigestOfDesiredDockerImage = dockerImageWithDigest.substring(hashStartingPositionInActualDockerImage + 1)
+                if (cachedDockerImageDigestAndDiskName.dockerImageDigest == actualDigestOfDesiredDockerImage) {
+                  true
+                } else {
+                  jobLogger.info(s"Cached Docker image digest mismatch. Requested docker image $dockerImageAsSpecifiedByUser has different digest than " +
+                    s"corresponding cached image located at the ${cachedDockerImageDigestAndDiskName.diskImageName} disk image. " +
+                    s"Digest of requested image is ${actualDigestOfDesiredDockerImage}, but digest of cached image is ${cachedDockerImageDigestAndDiskName.dockerImageDigest}. " +
+                    s"Docker image cache feature will not be used for this task.")
+
+                  false
+                }
+              } else {
+                jobLogger.error(s"Programmer error ! Odd docker image name where supposed to be name with digest: $dockerImageWithDigest")
+                false
+              }
```

Comment:
ToL: Are there existing unit test classes available that could exercise these code branches? It would keep folks from breaking or removing this functionality in the future.